M2/Macaulay2/m2/programs.m2

```diff
@@ -146,14 +146,16 @@ runProgram(Program, String, String) := opts -> (program, name, args) -> (
 	    makeDirectory opts.RunDirectory;
 	"cd " | opts.RunDirectory | " && " ) else "";
     cmd = cmd | program#"path" | addPrefix(name, program#"prefix") | " " | args;
+    if opts.Verbose then printerr("running: ", cmd);
     if match("\\|", cmd) then cmd = "{ " | cmd | ";}";
-    returnValue := run (cmd | " > " | outFile | " 2> " | errFile);
-    message := "running: " | cmd | "\n";
+    returnValue := (
+	if opts.Verbose
+	then run ("(((((" | cmd | "; echo $? >&4) | tee " | outFile |
+	    ") 3>&1 1>&2 2>&3 | tee " | errFile |
+	    " >&5) 4>&1) | (read r; exit $r)) 5>&1")
+	else run (cmd | " > " | outFile | " 2> " | errFile));
```

Comment:
This works great, thank you!
For future though, I wonder if it would be best to add a c++ routine for this.